src/components/text_truncate/text_truncate.tsx

```diff
@@ -138,9 +138,14 @@ const EuiTextTruncateWithWidth: FunctionComponent<
   // If necessary, wait a tick on mount before truncating
   const [ready, setReady] = useState(!calculationDelayMs);
   useEffect(() => {
+    let timerId: NodeJS.Timeout;
     if (calculationDelayMs) {
-      setTimeout(() => setReady(true), calculationDelayMs);
+      timerId = setTimeout(() => setReady(true), calculationDelayMs);
     }
+
+    return () => {
+      clearTimeout(timerId);
+    };
```

Comment:
Not a super huge deal, but would this also work to solve the issue without calling `clearTimeout` if no timeout has been set?

```suggestion
    if (calculationDelayMs) {
      const timerId = setTimeout(() => setReady(true), calculationDelayMs);
      return () => {
        clearTimeout(timerId);
      };
    }
```

Reply:
I ended up grabbing this minor cleanup while writing unit tests for this and confirming it still worked as expected (https://github.com/elastic/eui/pull/7495/commits/48c0e3e769312c679c83c91b30322ebd38d81322) - LMK if you foresee any possible issues with this change! If not, I think this PR is good to go! 🎉

Reply:
LGTM as both the async update and the cleanup are conditionally running, thanks for the change!